lomond/session.py

```diff
@@ -33,12 +35,19 @@ def __init__(self, websocket):
         self._lock = threading.Lock()
 
         self._sock = None
-        self._poll_start = time.time()
-        self._last_ping = time.time()
+        self._poll_start = None
+        self._next_ping = None
+        self._last_pong = None
+        self._start_time = None
 
     def __repr__(self):
         return "<ws-session '{}'>".format(self.websocket.url)
 
+    @property
+    def _time(self):
```

Comment:
This simplifies things for the ping_timeout calculation.

Essentially this is as clock that starts when the websocket is ready.